### PR TITLE
Added type checking for ExportedProgram

### LINKED ISSUE #116952: torch.export.save() should check input type.
### 🐛 Describe the bug

Currently if I don't pass in the input as ExportedProgram, there is no good error message telling me I'm doing something wrong.

At least we should do some type checking.

### Versions

main branch

cc @avikchaudhuri @gmagogsfm @tugsbayasgalan @angelayi @suo @ydwu4

### PR DESCRIPTION
**Pull request by RazaProdigy**
Fixes #116952

Added type checking for ExportedProgram in save function. Please review.

### CODE REVIEW

File: torch/export/__init__.py [hunk 1]
```diff
@@ -252,6 +252,9 @@ def forward(self, x):
     """
     from torch._export import save
 
+    if not isinstance(ep, ExportedProgram):
+        raise TypeError("The 'ep' parameter must be an instance of 'ExportedProgram'.")
```
Comment (avikchaudhuri) [inline torch/export/__init__.py:256]: Maybe include the type of `ep` in the error message as well? Like, expected this got that.

**Review by avikchaudhuri — APPROVED** [review]

File: torch/export/__init__.py (hunk 1 above)
Comment (RazaProdigy) [reply torch/export/__init__.py]: Should be fine now. 🙂

**Comment by zhxchen17** [comment]
@RazaProdigy thanks for the contribution! Mind fixing the lint error before we merge this?
```
|-        raise TypeError(f"The 'ep' parameter must be an instance of 'ExportedProgram', got '{type(ep).__name__}' instead.")
2024-01-22T20:47:04.0357974Z          256 |+        raise TypeError(
2024-01-22T20:47:04.0359372Z          257 |+            f"The 'ep' parameter must be an instance of 'ExportedProgram', got '{type(ep).__name__}' instead."
2024-01-22T20:47:04.0360570Z          258 |+        )
```
Otherwise it cannot the CI and not landable.

**Comment by RazaProdigy** [comment]
@zhxchen17 I fixed the lint error , it was a formatting issue from my end. 
Just curious, Are some checks expected to fail? Is it something to do with how the docker image is created?

**Comment by zhxchen17** [comment]
@RazaProdigy I looked at the failed tests and they seem to be not very related to your diff, assuming they are just flaky. I can try again.

**Comment by RazaProdigy** [comment]
> @RazaProdigy I looked at the failed tests and they seem to be not very related to your diff, assuming they are just flaky. I can try again.

Alright thank you so much!

**Pull request closed without merging**